### PR TITLE
Check for a valid path on any map

### PR DESCRIPTION
**Pull request by cdubz**
Implements isLegal() in Map() per Issue #1.

### CODE REVIEW

**Comment by cdubz** [comment]
Last three commits (3b097e1, 5ad3c86, 5ad3c86) are just a couple of piggy back changes to get rid of errors in the console and broken functionality caused by the updated React core from c26405d.

File: lib/Pathfinder.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+var PF = require('pathfinding');
+
+
+function Pathfinder(map) {
```
Comment (bovard) [inline lib/Pathfinder.js]: I don't think this needs to be a class anymore, you could change it to a function like https://github.com/bovard/raid/blob/master/lib/Utils/MoveUtils.js

Comment (bovard) [inline lib/Pathfinder.js]: maybe:

```
function findPath(terrain, start, end) {
    //log from above
}

function doesPathExist(terrain, start, end) {
      var path = findPath(terrain, start, end);
      if (path) {
          return true;
      }
     return false;
}

function nextDirTo(terrain, start, end) {
    var path = findPath(terrain, start, end);
    if (path) { return new Direction(something, something) }
   return null
}
```

**Comment by bovard** [comment]
Nice!

**Comment by cdubz** [comment]
:godmode:

**Pull request merged**